communicator/winrm/provisioner.go

```diff
@@ -0,0 +1,109 @@
+package winrm
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/mitchellh/mapstructure"
+)
+
+const (
+	// DefaultUser is used if there is no user given
+	DefaultUser = "Administrator"
+
+	// DefaultPort is used if there is no port given
+	DefaultPort = 5985
+
+	// DefaultScriptPath is used as the path to copy the file to
+	// for remote execution if not provided otherwise.
+	DefaultScriptPath = "C:/Temp/script.cmd"
```

Comment:
You'll probably want to reflect the tweak from #1588 for both the `winrm` and the `ssh` provisioners. See the linked issue for details.